Parsers/ASimWebSession/Parsers/vimWebSessionCiscoFirepower.yaml

```diff
@@ -0,0 +1,264 @@
+                              )
+          | project-rename DstUsername = DestinationUserName
+          | extend
+              DstUsernameType = _ASIM_GetUsernameType(DstUsername),
+              DstUserType = _ASIM_GetUserType(DstUsername, "");
+      let Filelogs = alllogs
+          | where DeviceEventClassID in ("File:500:1", "FileMalware:502:1")
+          | extend AdditionalFields = bag_pack(
+                                  "connectionInstance",
+                                  ProcessID,
+                                  "signaturedata",
+                                  DeviceCustomString4,
+                                  "RequestClientApplication",
+                                  RequestClientApplication
+                              );
+      union Fireamplogs, Filelogs
+      | invoke _ASIM_ResolveNetworkProtocol('Protocol')
+      | extend NetworkProtocol = iff(NetworkProtocol == "Unassigned" and Protocol !in (63, 68, 99, 114, 253, 254), Protocol, NetworkProtocol)
+      | lookup DirectionLookup on CommunicationDirection
+      | extend
+          EventStartTime = coalesce(unixtime_milliseconds_todatetime(start), unixtime_milliseconds_todatetime(tolong(ReceiptTime))),
+          DstIpAddr = coalesce(DestinationIP, DeviceCustomIPv6Address3),
+          SrcIpAddr = coalesce(SourceIP, DeviceCustomIPv6Address2),
+          EventSeverity = case(
+                      DvcAction == "Allow" and DeviceCustomString3 =~ "Malware",
+                      "High",
+                      DvcAction == "Deny" and DeviceCustomString3 =~ "Malware",
+                      "Medium",
+                      DvcAction == "Deny" and DeviceCustomString3 !~ "Malware",
+                      "Low",
+                      "Informational"
+                  ),
+          EventOriginalType = case(
+                          DeviceEventClassID has "File:500:1",
+                          "File Event",
+                          DeviceEventClassID has "FileMalware:502:1",
+                          "FileMalware Event",
+                          Activity
+                      ),
+          FileContentType = FileType,
+          HttpContentType = FileType,
+          FileSize = tolong(FileSize)
+      | extend Ip_device = iff(DeviceName matches regex "(([0-9]{1,3})\\.([0-9]{1,3})\\.([0-9]{1,3})\\.(([0-9]{1,3})))", DeviceName, "")
+      | extend
+          DvcIpAddr = Ip_device,
+          DeviceName = iff(isempty(Ip_device), DeviceName, "")
+      | extend host = coalesce(DeviceName, Computer)
+      | invoke _ASIM_ResolveDvcFQDN('host')
+      | extend 
+          EventCount = int(1),
+          EventSchema = "WebSession",
+          EventSchemaVersion = "0.2.6",
+          EventType = "HTTPsession"
+      | project-rename
+          EventVendor = DeviceVendor,
+          EventProduct = DeviceProduct,
+          EventProductVersion = DeviceVersion,
+          DstPortNumber = DestinationPort,
+          SrcUsername = SourceUserName,
```

Comment:
invalid value in SrcUsername
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/abd8cce0-6d3e-4a0c-9189-c44270f3365b)

Reply:
As discussed this value is coming in our environment from the source itself. But there might be chance that in customer's environment there will be valid values.